internal/wasm/call_context_test.go

```diff
@@ -288,3 +290,75 @@ func TestCallContext_CallDynamic(t *testing.T) {
 		require.False(t, ok, "expected no opened files")
 	})
 }
+
+func TestCallContext_SetExitCodeOnCanceledOrTimeout(t *testing.T) {
+	t.Run("timeout", func(t *testing.T) {
+		cc := &CallContext{Closed: new(uint64), module: &ModuleInstance{Name: "test"}}
+		const duration = time.Second
+		ctx, cancel := context.WithTimeout(context.Background(), duration)
+		defer cancel()
+		done := cc.SetExitCodeOnCanceledOrTimeout(ctx)
+		time.Sleep(duration * 2)
+		defer done()
+
+		err := cc.FailIfClosed()
+		require.EqualError(t, err, "module \"test\" closed with context deadline exceeded")
+	})
+	t.Run("cancel", func(t *testing.T) {
```

Comment:
maybe also a redundant call to cancel() to make sure nothing panics or otherwise gets weird